core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1930,7 +1930,20 @@ public static MutableRel unifyAggregates(MutableAggregate query,
       }
       final List<AggregateCall> aggregateCalls = new ArrayList<>();
       for (AggregateCall aggregateCall : query.aggCalls) {
-        if (aggregateCall.isDistinct()) {
+        if (aggregateCall.isDistinct() && aggregateCall.getArgList().size() == 1) {
+          int aggIndex = aggregateCall.getArgList().get(0);
+          if (target.groupSet.asSet().contains(aggIndex)) {
+            int newIndex = target.groupSet.asList().indexOf(aggIndex);
```

Comment:
We'd better use a variable to represent `target.groupSet.asSet()` in case we have to transform it again and again.

Reply:
ok,Thanks.